agent/pkg/singularity/singularity.go

```diff
@@ -387,6 +378,49 @@ func (s *SingularityClient) RunContainer(
 	}, nil
 }
 
+// Sets the environment of the process that will run the Singularity/apptainer command.
+func (s *SingularityClient) setCommandEnvironment(
+	req cproto.RunSpec, cudaVisibleDevices []string, cmd *exec.Cmd,
+) {
+	// Per https://pkg.go.dev/os/exec#Cmd.Env, if cmd.Env is nil, the new process uses the current
+	// process's environment. If this in not the case, for example because we specify something to
+	// control Singularity operation, then we need to specify everything needed from the current
+	// environment,
+	if req.DeviceType == device.CUDA {
+		cudaVisibleDevicesVar := strings.Join(cudaVisibleDevices, ",")
+		cmd.Env = append(cmd.Env,
+			fmt.Sprintf("SINGULARITYENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
+			fmt.Sprintf("APPTAINERENV_CUDA_VISIBLE_DEVICES=%s", cudaVisibleDevicesVar),
+		)
+	}
+	if req.DeviceType == device.ROCM {
+		// Avoid this problem: https://github.com/determined-ai/determined-ee/pull/922
+		// by not setting both ROCR_VISIBLE_DEVICES & CUDA_VISIBLE_DEVICES
+		s.addToCommandEnvironment(cmd, "ROCR_VISIBLE_DEVICES")
```

Comment:
Why are we not doing the same as on 392-393 -- i.e. 
SINGULARITYENV_ROCR_VISIBLE_DEVICES & APPTAINERENV_ROCR_VISIBLE_DEVICES?

Or switch these to use  s.addToCommandEnvironment(cmd,...)?

Seems like they all should be treated the same.

Reply:
Those are not variables that Singularity [documents](https://docs.sylabs.io/guides/latest/user-guide/gpu.html#) as being supported.

Reply:
OK.  Probably a comment indicating that --nv & --rocm support in Singularity/Apptainer requires special treatment of the variables.

Reply:
Bradley has a story to move us away from `--nv`, so likely this will be revised soon.